snuba/web/rpc/common/debug_info.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+
+from sentry_protos.snuba.v1.request_common_pb2 import (
+    QueryInfo,
+    QueryMetadata,
+    QueryStats,
+    ResponseMeta,
+    TimingMarks,
+)
+
+from snuba.utils.metrics.timer import Timer
+from snuba.web import QueryResult
+
+
+def extract_response_meta(
+    request_id: str, debug: bool, query_results: List[QueryResult], timers: List[Timer]
+) -> ResponseMeta:
+    query_info = []
+
+    if debug:
+        for query_result, timer in zip(query_results, timers):
+            stats = query_result.extra.get("stats", {}) or {}
+            profile = query_result.result.get("profile", {}) or {}
+
+            timer_data = timer.for_json()
+            timing_marks = TimingMarks(
+                marks_ms=timer_data.get("marks_ms", {}),
+                duration_ms=int(timer_data.get("duration_ms", 0)),
+                timestamp=int(timer_data.get("timestamp", 0)),
+                tags=timer_data.get("tags", {}),
+            )
+
+            query_stats = QueryStats(
+                rows_read=stats.get("result_rows", 0),
+                columns_read=stats.get("result_cols", 0),
+                blocks=profile.get("blocks", 0),
+                progress_bytes=profile.get("progress_bytes", 0),
+                max_threads=stats.get("quota_allowance", {})
+                .get("summary", {})
+                .get("threads_used"),
```

Comment:
Should the default value of `max_threads` be `0` (or a valid number) in case `stats.get("quota_allowance").get("summary").get("threads_used")` is not found / resolves to `None`? It looks like as is this will resolve to `None`.

Reply:
In Protobuf, the handling of None and 0 for numeric fields differs from standard Python behavior. If nested get() calls result in None, the max_threads field in the Protobuf message is automatically set to 0.